frost-core/src/frost/keys.rs

```diff
@@ -142,7 +144,27 @@ where
 /// This is a (public) commitment to one coefficient of a secret polynomial used for performing
 /// verifiable secret sharing for a Shamir secret share.
 #[derive(Clone, Copy, PartialEq)]
-pub(super) struct CoefficientCommitment<C: Ciphersuite>(pub(super) Element<C>);
+pub struct CoefficientCommitment<C: Ciphersuite>(pub(crate) Element<C>);
+
+impl<C> CoefficientCommitment<C>
+where
+    C: Ciphersuite,
+{
+    /// Creates a new commitment from a coefficient input
+    pub fn deserialize(
+        coefficient: <C::Group as Group>::Serialization,
+    ) -> Result<CoefficientCommitment<C>, Error<C>> {
+        Ok(Self(
+            <C::Group as Group>::deserialize(&coefficient)
+                .map_err(|_| Error::from(GroupError::MalformedElement))?,
```

Comment:
You can resurface the original error:

```suggestion
            <C::Group as Group>::deserialize(&coefficient).map_err(Error::from)?,
```